consent/manager_test_helpers.go

```diff
@@ -0,0 +1,513 @@
+				ID:              "rev-session-1",
+				AuthenticatedAt: time.Now(),
+				Subject:         "subject-1",
+			}))
+
+			require.NoError(t, m.CreateAuthenticationSession(context.TODO(), &AuthenticationSession{
+				ID:              "rev-session-2",
+				AuthenticatedAt: time.Now(),
+				Subject:         "subject-2",
+			}))
+
+			require.NoError(t, m.CreateAuthenticationSession(context.TODO(), &AuthenticationSession{
+				ID:              "rev-session-3",
+				AuthenticatedAt: time.Now(),
+				Subject:         "subject-1",
+			}))
+
+			for i, tc := range []struct {
+				subject string
+				ids     []string
+			}{
+				{
+					subject: "subject-1",
+					ids:     []string{"rev-session-1", "rev-session-3"},
+				},
+				{
+					subject: "subject-2",
+					ids:     []string{"rev-session-1", "rev-session-3", "rev-session-2"},
+				},
+			} {
+				t.Run(fmt.Sprintf("case=%d/subject=%s", i, tc.subject), func(t *testing.T) {
+					require.NoError(t, m.RevokeUserAuthenticationSession(context.TODO(), tc.subject))
+
+					for _, id := range tc.ids {
+						t.Run(fmt.Sprintf("id=%s", id), func(t *testing.T) {
+							_, err := m.GetAuthenticationSession(context.TODO(), id)
+							assert.EqualError(t, err, pkg.ErrNotFound.Error())
+						})
+					}
+				})
+			}
+		})
+
+		t.Run("case=revoke-handled-consent-request", func(t *testing.T) {
+			cr1, hcr1 := MockConsentRequest("rv1", false, 0, false, false, false)
+			cr2, hcr2 := MockConsentRequest("rv2", false, 0, false, false, false)
+			clientManager.CreateClient(context.TODO(), cr1.Client)
+			clientManager.CreateClient(context.TODO(), cr2.Client)
+
+			require.NoError(t, m.CreateConsentRequest(context.TODO(), cr1))
+			require.NoError(t, m.CreateConsentRequest(context.TODO(), cr2))
+			_, err := m.HandleConsentRequest(context.TODO(), "challengerv1", hcr1)
+			require.NoError(t, err)
+			_, err = m.HandleConsentRequest(context.TODO(), "challengerv2", hcr2)
+			require.NoError(t, err)
+
+			fositeManager.CreateAccessTokenSession(nil, "trva1", &fosite.Request{ID: "challengerv1", RequestedAt: time.Now()})
+			fositeManager.CreateRefreshTokenSession(nil, "rrva1", &fosite.Request{ID: "challengerv1", RequestedAt: time.Now()})
+			fositeManager.CreateAccessTokenSession(nil, "trva2", &fosite.Request{ID: "challengerv2", RequestedAt: time.Now()})
+			fositeManager.CreateRefreshTokenSession(nil, "rrva2", &fosite.Request{ID: "challengerv2", RequestedAt: time.Now()})
+
+			for i, tc := range []struct {
+				subject string
+				client  string
+				at      string
+				rt      string
+				ids     []string
+			}{
+				{
+					at: "trva1", rt: "rrva1",
+					subject: "subjectrv1",
+					client:  "",
+					ids:     []string{"challengerv1"},
+				},
+				{
+					at: "trva2", rt: "rrva2",
+					subject: "subjectrv2",
+					client:  "clientrv2",
+					ids:     []string{"challengerv2"},
+				},
+			} {
+				t.Run(fmt.Sprintf("case=%d/subject=%s", i, tc.subject), func(t *testing.T) {
```

Comment:
This is the only test that was modified. Since we don't have access to the actual fosite memory manager anymore, we rely on the interface access methods to validate the Access Tokens/Refresh Tokens are present or not